client/my-sites/controller.js

```diff
@@ -109,7 +109,7 @@ module.exports = {
 	 * Set up site selection based on last URL param and/or handle no-sites error cases
 	 */
 	siteSelection( context, next ) {
-		const siteID = route.getSiteFragment( context.path );
+		const siteID = context.params.site_id;
```

Comment:
Ah cool, that makes sense.

@mtias: Looks like you introduced `getSiteFragment` here - mind taking a look at this?